pkgs/development/lua-modules/lib.nix

```diff
@@ -89,58 +90,50 @@ rec {
     , rocksSubdir
     }: let
       rocksTrees = lib.imap0
-        (i: dep: "{ name = [[dep-${toString i}]], root = '${dep}', rocks_dir = '${dep}/${dep.rocksSubdir}' }")
+        (i: dep: { name = "dep-${toString i}"; root = "${dep}"; rocks_dir = "${dep}/${dep.rocksSubdir}"; })
         requiredLuaRocks;
 
       # Explicitly point luarocks to the relevant locations for multiple-output
       # derivations that are external dependencies, to work around an issue it has
       # (https://github.com/luarocks/luarocks/issues/766)
-      depVariables = lib.concatMap ({name, dep}: [
-        "${name}_INCDIR='${lib.getDev dep}/include';"
-        "${name}_LIBDIR='${lib.getLib dep}/lib';"
-        "${name}_BINDIR='${lib.getBin dep}/bin';"
-      ]) externalDeps';
+      depVariables = zipAttrsWithLast (lib.lists.map ({name, dep}: {
+        "${name}_INCDIR" = "${lib.getDev dep}/include";
+        "${name}_LIBDIR" = "${lib.getLib dep}/lib";
+        "${name}_BINDIR" = "${lib.getBin dep}/bin";
+      }) externalDeps');
+      zipAttrsWithLast = lib.attrsets.zipAttrsWith (name: lib.lists.last);
 
       # example externalDeps': [ { name = "CRYPTO"; dep = pkgs.openssl; } ]
       externalDeps' = lib.filter (dep: !lib.isDerivation dep) externalDeps;
 
       externalDepsDirs = map
-        (x: "'${builtins.toString x}'")
+        (x: builtins.toString x)
         (lib.filter (lib.isDerivation) externalDeps);
-
-      extraVariablesStr = lib.concatStringsSep "\n "
-        (lib.mapAttrsToList (k: v: "${k}='${v}';") extraVariables);
   in ''
```

Comment:
why not call toLua here at the upper level ?

Reply:
These lines are removed :slightly_smiling_face: in upper level we do:
```nix
"variables = ${toLua {} (depVariables // extraVariables)}"
```

Reply:
Sorry. I think I misread where comment were attached to. At the moment `toLua` can only produce single Lua value and have no configuration to produce Lua content with setting multiple variables. E.g. result of 
```nix
toLua {} { variables = depVariables // extraVariables; }
```
would be something like
```lua
{
  ["variables"] = {
    ["CRYPTO_BINDIR"] = "/nix/store/05qna7girdhnzx38jkhr4yyh7gmnizv3-openssl-3.0.8-bin/bin",
    ["CRYPTO_INCDIR"] = "/nix/store/ggsmy3rwsr6q6ri45ck3jdqnz9cf6dm6-openssl-3.0.8-dev/include",
    ["CRYPTO_LIBDIR"] = "/nix/store/s8vg2h8xzqmjd72f3g3p1jqy2lbbapc6-openssl-3.0.8/lib",
    ["OPENSSL_BINDIR"] = "/nix/store/05qna7girdhnzx38jkhr4yyh7gmnizv3-openssl-3.0.8-bin/bin",
    ["OPENSSL_INCDIR"] = "/nix/store/ggsmy3rwsr6q6ri45ck3jdqnz9cf6dm6-openssl-3.0.8-dev/include",
    ["OPENSSL_LIBDIR"] = "/nix/store/s8vg2h8xzqmjd72f3g3p1jqy2lbbapc6-openssl-3.0.8/lib"
  }
}
```

And I'm not that familiar with Lua to know how to turn table value into bindings for local/global variables via Lua code.

Reply:
Added configuration `asBindings` for `toLua` and lifted all calls to top expression. Note that comments no longer present in generated `luarocks-config.lua`. E.g. for `nix build '.#luaPackages.cqueues'` I get now:

```
external_deps_dirs = {
  "/nix/store/yrb14jdd2wh1xrzz4vlbsdn21vkvimsd-wrap-lua-hook",
  "/nix/store/kp355cm1wyv8dbx6rq79n823iarrq061-luarocks-3.9.1",
  "/nix/store/pi7ca6hpy4g82l96f9d8yk2k69q9wr0l-lua-5.2.4",
  "/nix/store/ggsmy3rwsr6q6ri45ck3jdqnz9cf6dm6-openssl-3.0.8-dev"
}
local_cache = ""
rocks_subdir = "cqueues-20200726-0-rocks"
rocks_trees = {
  {
    ["name"] = "current",
    ["rocks_dir"] = "current",
    ["root"] = "/nix/store/3bfvabk59yp30apbvsmnf1cq2ks9cswb-lua20200726.52-cqueues-20200726-0"
  }
}
variables = {
  ["CRYPTO_BINDIR"] = "/nix/store/05qna7girdhnzx38jkhr4yyh7gmnizv3-openssl-3.0.8-bin/bin",
  ["CRYPTO_INCDIR"] = "/nix/store/ggsmy3rwsr6q6ri45ck3jdqnz9cf6dm6-openssl-3.0.8-dev/include",
  ["CRYPTO_LIBDIR"] = "/nix/store/s8vg2h8xzqmjd72f3g3p1jqy2lbbapc6-openssl-3.0.8/lib",
  ["OPENSSL_BINDIR"] = "/nix/store/05qna7girdhnzx38jkhr4yyh7gmnizv3-openssl-3.0.8-bin/bin",
  ["OPENSSL_INCDIR"] = "/nix/store/ggsmy3rwsr6q6ri45ck3jdqnz9cf6dm6-openssl-3.0.8-dev/include",
  ["OPENSSL_LIBDIR"] = "/nix/store/s8vg2h8xzqmjd72f3g3p1jqy2lbbapc6-openssl-3.0.8/lib"
}



```